tests/units/anta_tests/__init__.py

```diff
@@ -17,17 +50,37 @@ def test(device: AntaDevice, data: dict[str, Any]) -> None:
     See `tests/units/anta_tests/README.md` for more information on how to use it.
     """
     # Instantiate the AntaTest subclass
+    if "inputs" not in data:
+        data["inputs"] = None
     test_instance = data["test"](device, inputs=data["inputs"], eos_data=data["eos_data"])
     # Run the test() method
     asyncio.run(test_instance.test())
     # Assert expected result
     assert test_instance.result.result == data["expected"]["result"], f"Expected '{data['expected']['result']}' result, got '{test_instance.result.result}'"
     if "messages" in data["expected"]:
         # We expect messages in test result
-        assert len(test_instance.result.messages) == len(data["expected"]["messages"])
+        assert len(test_instance.result.messages) == len(
+            data["expected"]["messages"]
+        ), f"Expected {len(data['expected']['messages'])} messages, got {len(test_instance.result.messages)}"
         # Test will pass if the expected message is included in the test result message
         for message, expected in zip(test_instance.result.messages, data["expected"]["messages"]):  # NOTE: zip(strict=True) has been added in Python 3.10
             assert expected in message
     else:
         # Test result should not have messages
-        assert test_instance.result.messages == []
+        assert test_instance.result.messages == [], "There are untested messages"
```

Comment:
not necessarily true - there could be messages because the test failed when it should have succeeded. 

```suggestion
        assert test_instance.result.messages == [], "There are unexpected messages"
```